src/shared.rs

```diff
@@ -13,7 +13,7 @@ pub const CONTROL_PORT: u16 = 7835;
 #[derive(Serialize, Deserialize)]
 pub enum ClientMessage {
     /// Initial client message specifying a port to forward.
-    Hello(u16),
+    Hello((u16, Option<String>)),
```

Comment:
This can just be a standard `Hello(u16, Option<String>)` variant, doesn't need the nested tuple!